pkg/platform/runtime/buildscript/file.go

```diff
@@ -22,40 +22,59 @@ type projecter interface {
 	Name() string
 }
 
-func NewScriptFromProject(proj projecter, auth *authentication.Auth) (*Script, error) {
-	return newScriptFromFile(filepath.Join(proj.ProjectDir(), constants.BuildScriptFileName), proj.Owner(), proj.Name(), auth)
-}
+var ErrBuildscriptNotExist = errors.New("Build script does not exist")
 
-func newScriptFromFile(path, org, project string, auth *authentication.Auth) (*Script, error) {
-	if data, err := fileutils.ReadFile(path); err == nil {
-		return NewScript(data)
-	} else if !errors.Is(err, os.ErrNotExist) {
-		return nil, errs.Wrap(err, "Could not read build script")
-	}
+// ScriptFromProjectWithFallback will source the buildscript from the project, and create it if it does not exist.
+func ScriptFromProjectWithFallback(proj projecter, auth *authentication.Auth) (*Script, error) {
```

Comment:
Hopefully this naming makes it evident that this is doing more than the name would otherwise suggest.